lib/ansible/modules/cloud/amazon/ec2_group.py

```diff
@@ -511,6 +511,10 @@ def main():
                     rule['from_port'] = None
                     rule['to_port'] = None
 
+                # rule may have empty cidr_id/group_id/group_name in which case a real rule won't be created
+                if not (rule.get('group_id') and rule.get('group_name') and rule.get('cidr_ip')):
```

Comment:
This doesn't seem right to me. This looks like it prevents a rule being created if *any* of them are not set, rather than your intention of if none of them are set. Changing the `and`s to `or`s should do what you mean.